### PR TITLE
chore(ci): refactor gh actions to use separate file for different envs

### PR DESCRIPTION
**Pull request by clemsos**
# Description

The current structure of gh actions workflows is confusing, and the use of the `workflow_run` condition makes it even more so. The current PR aims at simplifying things by splitting into 3 main files for pr, master and prod describing what should happen in each context.

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  run-all-tests:
+    uses: ./.github/workflows/_tests.yml
+
+  deploy-locksmith-staging:
```
Comment (julien51) [inline .github/workflows/master.yml:12]: We should also deploy all netlify services, correct?

File: .github/workflows/production.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+name: "Production Branch"
+
+on:
+  push:
+    branches:
+      - production
+
+jobs:
+  run-all-tests:
+    uses: ./.github/workflows/_tests.yml
```
Comment (julien51) [inline .github/workflows/production.yml:10]: Same as above?

File: .github/workflows/pull-request.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    needs: run-tests
+    uses: ./.github/workflows/_netlify.yml
+    with:
+      service: unlock-protocol-com
+      target-env: staging
+    secrets:
+      SITE_ID: ${{ secrets.UNLOCK_PROTOCOL_COM_NETLIFY_STAGING_AUTH_TOKEN }}
+      AUTH_TOKEN: ${{ secrets.UNLOCK_PROTOCOL_COM_NETLIFY_STAGING_AUTH_KEY }}
+  
+  # deploy-paywall-app
```
Comment (julien51) [inline .github/workflows/pull-request.yml:53]: ha! let's make sure we have these :)

**Review by julien51 — COMMENTED** [review]
ok I like this new approach. Do you want to add the tasks to deploy all front-ends too?

They should be pretty similar?

File: .github/workflows/master.yml (hunk 1 above)
Comment (clemsos) [reply .github/workflows/master.yml:12]: yep, fixing the workflow for PRs first, then will add

File: .github/workflows/_netlify.yml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      - name: Restore docker
+        run: .github/actions/docker-cache/restore.sh ${{ runner.temp }}/docker_cache/cache.tar
+      - name: "Export secrets to env"
+        id: filter_secrets
+        run: | 
+          SERVICE_NAME="${{ inputs.service }}"
+          UPCASE_SERVICE="${SERVICE_NAME^^}"
+          PREFIX="${UPCASE_SERVICE//-/_}_NETLIFY_${{ inputs.target-env }}"
+          PREFIX="${PREFIX^^}" # uppercase
+          echo "${PREFIX}"
```
Comment (julien51) [inline .github/workflows/_netlify.yml:90]: 🧹

**Review by julien51 — APPROVED** [review]
Still looks good

File: .github/workflows/_netlify.yml [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      SITE_ID:
+        required: true
+      AUTH_TOKEN:
+        required: true
+      # unlock-app only
+      BASE64_WEDLOCKS_PUBLIC_KEY:
+        required: false
+      STRIPE_KEY:
+        required: false
+      UNLOCK_GA_ID:
```
Comment (clemsos) [inline .github/workflows/_netlify.yml:30]: do we need to keep that as an env var?

**Pull request merged**